hamilton/plugins/pandas_extensions.py

```diff
@@ -82,43 +83,256 @@ def save_data(self, data: DATAFRAME_TYPE) -> Dict[str, Any]:
 
 
 @dataclasses.dataclass
-class CSVDataAdapter(DataFrameDataLoader):
-    """Data loader for CSV files. Note that this currently does not support the wide array of
-    data loading functionality that pandas does. We will be adding this in over time, but for now
-    you can subclass this or open up an issue if this doesn't have what you want.
-
-    Note that, when saving, this does not currently save the index.
-    We'll likely want to enable this in the future as an optional subclass,
-    in which case we'll separate it out.
+class PandasCsvReader(DataLoader):
+    """
+    Class that handles saving CSV files with pandas.
+    Maps to https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html
     """
 
-    path: str
```

Comment:
why don't we keep it as `path` for backwards compatibility? I know this now wont map 1-1 with pandas, but I think that's fine/it's clear enough. We could add a bunch of extra validation and have both options, and name one as deprecated but that seems like more trouble than it's worth. We could just note that it's called `path` for legacy reasons.